tsconfig.base.json

```diff
@@ -24,6 +24,7 @@
       // includes support for browser APIs
       "dom"
     ],
+    "skipLibCheck": true,
```

Comment:
`resize-observer-polyfill`, via `eui.d.ts`, exported `ResizeObserver` without exporting `ResizeObserverCallback`. The polyfill also exported `contentRect` which collided with typescript's built-in one.

Reply:
This impacts all declaration files though, right? Can we address this individually instead of skipping the type checking for everything? https://www.typescriptlang.org/tsconfig#skipLibCheck

Reply:
This is temporary, until I fold in.

* My original solution, for the folded-in version, uses `@juggle/resize-observer` but after this PR, I have changed my mind because it is a ponyfill. 
* I have identified `@rkesters/resize-observer-polyfill` and `@4lolo/resize-observer-polyfill` as possible options. Since part of the conflict is caused by `@types/resize-observer-browser`, I have to check if they work.
* My worst case scenario involves writing my own type-def to be employed via `compilerOptions`.
* I remember hearing about some possible effort to upgrade typescript; that would impact change the direction completely.

Reply:
Okay that makes sense. Just as long as we have a plan.

Yes, I upgraded TypeScript to 4.1 in #1230 but I haven't gotten around to resolving the conflicts/test failures yet.